### PR TITLE
WIP Migration and dependencies

### PR DESCRIPTION
**Pull request by jalavik**
Adds a migrator manager which helps with database/index migrations.

Also adds some new temporary forks of invenio packages required.

Adds upgrade recipe for citations modules.

### CODE REVIEW

**Comment by kaplun** [comment]
Woah! What's this project? Can you explain it to me IRL? Sounds cool!

File: inspire/modules/migrator/manage.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                action='append',
+                default=None,
+                help='Specific collections to migrate.')
+@manager.option('--input', '-f', dest='file_input',
+                help='Specific collections to migrate.')
+def populate(records, collections, file_input=None):
+    """Train a set of records from the command line.
+
+    Usage: inveniomanage predicter train -r /path/to/json -o model.pickle
+    """
```
Comment (kaplun) [inline inspire/modules/migrator/manage.py]: Amend docstring :tongue:

File: inspire/modules/migrator/manage.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    if collections:
+        print("Migrating collections: {0}".format(",".join(collections)))
+
+    if file_input and not os.path.isfile(file_input):
+        print("{0} is not a file!".format(file_input), file=sys.stderr)
+        return
+
+    if file_input:
+        print("Migrating records from file: {0}".format(file_input))
+        # FIXME: Add hook to allow for pre-allocation of IDs (--force)
```
Comment (jalavik) [inline inspire/modules/migrator/manage.py]: @kaplun @Panos512 Here you can add the signal hook discussed in #379. (perhaps by adding a `force` flag above). Use `action=store_true` for flags in the `manager.option`.

File: inspire/modules/migrator/manage.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        db.engine.execute("SET FOREIGN_KEY_CHECKS=0;")
+
+        # Grab any table with foreign keys to bibrec for truncating
+        inspector = reflection.Inspector.from_engine(db.engine)
+        for table_name in inspector.get_table_names():
+            for foreign_key in inspector.get_foreign_keys(table_name):
+                if foreign_key["referred_table"] == "bibrec":
+                    tables_to_truncate.append(table_name)
+
+        if not prompt_bool("Going to truncate: {0}".format(
```
Comment (jalavik) [inline inspire/modules/migrator/manage.py]: @jmartinm Added a prompt here :)

Comment (kaplun) [inline inspire/modules/migrator/manage.py]: Can I type there `Yes, I know!`? :smile:

Comment (jalavik) [inline inspire/modules/migrator/manage.py]: It's the industry standard y/N. If you wanna implement it.. :-P

Comment (jacquerie) [inline inspire/modules/migrator/manage.py]: please_no

**Pull request merged**